lib/publiccloud/instance.pm

```diff
@@ -39,38 +39,44 @@ Timeout can be set by C<timeout> or 90 sec by default.
 C<<proceed_on_failure=>1>> allows to proceed with validation when C<cmd> is
 failing (return non-zero exit code)
 By default, the command is passed in single quotes to SSH.
-To avoid quoting us C<<no_quote=>1>>.
+To avoid quoting use C<<no_quote=>1>>.
 With C<<ssh_opts=>'...'>> you can overwrite all default ops which are in
 C<<$instance->ssh_opts>>.
 Use argument C<username> to specify a different username then
 C<<$instance->username()>>.
 Use argument C<rc_only> to only check for the return code of the command.
+Use argument C<cmd_timeout> to specify a soft timeout for the command. This must be lower than C<timeot>-1 otherwise it will be set to that value.
 =cut
 sub run_ssh_command {
-    my ($self, %args) = @_;
+    my $self = shift;
+    my %args = testapi::compat_args({cmd => undef}, ['cmd'], @_);
     die('Argument <cmd> missing') unless ($args{cmd});
     $args{ssh_opts} //= $self->ssh_opts() . " -i '" . $self->ssh_key . "'";
     $args{username} //= $self->username();
     $args{timeout}  //= SSH_TIMEOUT;
     $args{quiet}    //= 1;
     $args{no_quote} //= 0;
-    my $rc_only = $args{rc_only} // 0;
+    my $rc_only     = $args{rc_only}     // 0;
+    my $cmd_timeout = $args{cmd_timeout} // 0;
```

Comment:
did you found a place where we would break old code, when introduce `if ($args{timeout} > 0) { $cmd_timeout = $args{timeout}; $args{timeout}+=3;}` by default?
Or is there a other reason to not using `cmd_timeout` always?

Reply:
I wanted to prevent unexpected side-effects, but since you suggest to use `$args{timeout}` directly, I changes the behaviour accordingly.